presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java

```diff
@@ -302,4 +312,87 @@ public static Block extractMaskBlock(int maskChannel, Page page)
         }
         return maskBlock;
     }
+
+    public static AccumulatorFactory bind(JavaAggregationFunctionImplementation javaAggregationFunctionImplementation,
+                                          List<Integer> argumentChannels,
+                                          Optional<Integer> maskChannel,
+                                          List<Type> sourceTypes,
+                                          List<Integer> orderByChannels,
+                                          List<SortOrder> orderings,
+                                          PagesIndex.Factory pagesIndexFactory,
+                                          boolean distinct,
+                                          JoinCompiler joinCompiler,
+                                          List<LambdaProvider> lambdaProviders,
+                                          boolean spillEnabled,
+                                          Session session,
+                                          StandaloneSpillerFactory standaloneSpillerFactory)
+    {
+        if (javaAggregationFunctionImplementation instanceof BuiltInAggregationFunctionImplementation) {
+            return ((BuiltInAggregationFunctionImplementation) javaAggregationFunctionImplementation).bind(
```

Comment:
The only reason that you need an if instanceof here is the existence of `LazyAccumulatorFactoryBinder` which is not really needed. If you delete that, you can actually remove `AccumulatorFactoryBinder` from `JavaAggregationFunctionImplementation`, and create the binder class and call bind all here (the else block) for all `JavaAggregationFunctionImplementation`